src/mcp/mcp_client.py

```
@@ -0,0 +1,379 @@
+
+            if all_success:
+                # Refresh authorization status
+                self.mcp_connection_status = self._check_mcp_authorization()
+                
+            return all_success
+        except Exception:
+            return False
+
+    @staticmethod
+    def _create_cache_key(mcp_name: str) -> str:
+        return hashlib.md5(mcp_name.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)